### DISCUSSION TITLE
Create variables from strings ?

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#1948] [Ideas]
I'm needing to create unique variable names pragmatically and wondered if it's possible to get something like the ability to create variable names from strings as in Python or JS ?

I'm making am identical playlist source multiple times (same var name) , but it creates a skip function once for all - x.skip already exists , so when i skip one instance of the source all skip. They also restart each time a new instance of source is started.

It would be very useful.

**Comment by Russsgithub** [comment]
I think i may have been wrong about this, .. I add unique id's for each source, and now have control over them, I think (brief test).

**Comment by toots** [comment]
I've got some ongoing work that potentially could lead to the equivalent to javascript's eval function:
```ruby
x = eval("<liquidsoap script>")
```
Keep an eye on https://github.com/savonet/liquidsoap/pull/1934 for more details!
